.github/workflows/publish-github-pages.yml

```diff
@@ -0,0 +1,21 @@
+name: Publish to GitHub Pages
+on:
+    push:
+        branches:
+            - "main"
+jobs:
+    publish:
+        runs-on: ubuntu-latest
+        steps:
+            - name: Check out
+              uses: actions/checkout@v2
+            - name: Install modules
+              run: yarn
+            - name: Generate documentation
+              run: yarn run typedoc
+            - name: Publish to GitHub pages
+              uses: tsunematsu21/actions-publish-gh-pages@v1.0.1
+              with:
+                  dir: docs
+                  branch: gh-pages
+                  token: ${{ secrets.GH_PAGES_PAT }}
```

Comment:
@nebbles just for your reference here is how the documentation is generated for the sensirion package.